CNF-13832 Metadata apiv2 (#108)
openshift/kube-compare: docs/reference-config-guide-v2.md

required CRs are included in the input configuration then all required CRs
in the group are expected to be included and any which are missing will be reported. If none of
the required CRs in the group are included then no report of "missing content" for the group will be generated.

In this version Parts only help with organization of the components into groups and don't have any affection on the diff
process.

Thus, the file `metadata.yaml` includes an array denoted by `parts` of one or more objects. Each object includes:

- a key "name" with a value of a string typically identifying a workload or a set of workloads
- a key "components" defined as an array of objects

```yaml
# Every part denotes typically denotes a workload or set of workloads
apiVersion: v2 # Required will default to v1
parts:
  - name: ExamplePart1
    components:
      - name: ExampleComponent1
        ---- here goes ExampleComponent1 configuration ----
  - name: ExamplePart2
    ---- here goes Part2 configuration ----
```

"components" array includes objects with the following keys:

- a pair at key "name" with a value of a string identifying a component required for the part
- one of the following:

>   - `allOf`: Will cause a validation error if not all templates are matched for required templates.

Reviewer: ```suggestion
  - `allOf`: Will cause a validation error if not all the templates are matched. Use for sets of required templates.
```

>   - `allorNoneOf`: Will cause a validation error when one or more but not all templates are matched for optional blocks of templates.

Reviewer: ```suggestion
  - `allOrNoneOf`: Will cause a validation error when some but not all of the templates are matched. Use for sets of templates where the entire set is optional, but if at least one is present then all the others must also be present.
```

>   - `anyOf`: Will never cause a vaildation error, for completely optional templates

Reviewer: ```suggestion
  - `anyOf`: Will only cause validation errors for the content of matched templates. Use for completely optional templates.
```

>   - `noneOf`: Will cause a vaildation error if any of the templates are matched useful to exclude CRs

Reviewer: ```suggestion
  - `noneOf`: Will cause a validation error if any of the templates are matched. Use for excluding CRs.
```

>   - `oneOf`: Will cause a vaildation error when None or more than one templates is matched, for when you want to match exactly one template in the block

Reviewer: ```suggestion
  - `oneOf`: Will cause a validation error when none or more than one of the templates is matched. For requiring exactly one template in the set.
```

>   - `anyOneOf`: Will throw a validation error if more than one is matched, for when you optionally want only one of the templates the block

Reviewer: ```suggestion
  - `anyOneOf`: Will throw a validation error if more than one template is matched. Use for optionally allowing only one of the templates in the set.
```

```yaml

> # requiredTemplates contains all needed reference CRs to comply with ExampleComponent1

Reviewer: I think this comment is wrong now?
